app/routes/_gcn.circulars.$circularId/route.tsx

```diff
@@ -10,44 +10,56 @@ import {
   type LoaderFunctionArgs,
   json,
 } from '@remix-run/node'
-import { Link, useLoaderData } from '@remix-run/react'
+import { Link, Outlet, useLoaderData, useMatches } from '@remix-run/react'
 import { Button, ButtonGroup, Icon } from '@trussworks/react-uswds'
 
-import { get } from '../_gcn.circulars/circulars.server'
-import { PlainTextBody } from './Body'
-import { FrontMatter } from './FrontMatter'
-import { origin } from '~/lib/env.server'
+import { getUser } from '../_gcn._auth/user.server'
+import type { Circular } from '../_gcn.circulars/circulars.lib'
+import { getVersions, moderatorGroup } from '../_gcn.circulars/circulars.server'
+import { feature, origin } from '~/lib/env.server'
 import { getCanonicalUrlHeaders, pickHeaders } from '~/lib/headers.server'
 import { useSearchString } from '~/lib/utils'
-import type { BreadcrumbHandle } from '~/root/Title'
+import { useFeature } from '~/root'
 
-export const handle: BreadcrumbHandle<typeof loader> = {
-  breadcrumb({ data }) {
-    if (data) {
-      const { circularId, subject } = data
-      return `${circularId}: ${subject}`
-    }
-  },
-}
-
-export async function loader({ params: { circularId } }: LoaderFunctionArgs) {
+export async function loader({
+  params: { circularId },
+  request,
+}: LoaderFunctionArgs) {
   if (!circularId)
     throw new Response('circularId must be defined', { status: 400 })
-  const result = await get(parseFloat(circularId))
-  return json(result, {
-    headers: getCanonicalUrlHeaders(
-      new URL(`/circulars/${circularId}`, origin)
-    ),
-  })
+
+  let history: number[] = []
+  if (feature('CIRCULAR_VERSIONS')) {
+    history = await getVersions(parseFloat(circularId))
+    history.reverse() // Sort by descending version
+  }
+  const user = await getUser(request)
+  const userIsModerator = user?.groups.includes(moderatorGroup)
+  return json(
+    { circularId, history, userIsModerator },
+    {
+      headers: getCanonicalUrlHeaders(
+        new URL(`/circulars/${circularId}`, origin)
+      ),
+    }
+  )
 }
 
 export const headers: HeadersFunction = ({ loaderHeaders }) =>
   pickHeaders(loaderHeaders, ['Link'])
 
 export default function () {
-  const { circularId, body, bibcode, ...frontMatter } =
+  const { circularId, history, userIsModerator } =
     useLoaderData<typeof loader>()
   const searchString = useSearchString()
+
+  const routeMatch = useMatches().find(
+    (x) => x.id == 'routes/_gcn.circulars.$circularId.($version)'
+  )
+
+  const versionRouteMatchData = routeMatch?.data as Circular
+  const bibcode = versionRouteMatchData.bibcode
+  const version = versionRouteMatchData.version
```

Comment:
I'm apprehensive of components using route params from child routes. The other way around, a route using params from parents, I think is fine.

Reply:
Not sure I follow how that would work in this case. The getter for the specific circular only exists in the child route, so the bibcode and version aren't available at the parent level

Reply:
Right, so this UI must be in the _child_ route.

Reply:
These correspond to the buttons on top (text/json and citation links). These were the ones we talked about previously that I had moved below, Should I move them into the child component then?
![image](https://github.com/nasa-gcn/gcn.nasa.gov/assets/11023698/0284d551-d891-4951-bbd2-803e182bf2e0)

Reply:
Whatever order the button bar and the front matter appear in on the page, I think that they are both going to have to be in the child route because both require knowledge of the current version.

I would like you to have the parent route's loader return the list of versions, and employ `useRouteLoaderData` in the child route to retrieve the list. The parent route might end up having no UI content itself.

Reply:
Ahhh i see what you're saying. Okay i can make that work